### PR TITLE
stage damage performance and alternative comparison report fixes

### PR DESCRIPTION
**Pull request by rnugent3**
this PR contains fixes to alternative comparison report and fixes for stage damage performance

### CODE REVIEW

**Comment by rnugent3** [comment]
Parallelizing at the structure level cut a big inventory compute by half, but what I see in the profiling suggests that there is a lot more juice to squeeze.

**Comment by rnugent3** [comment]
the juice has been squeezed.... @Brennan1994 this is ready for your review. I'll be in the office tomorrow, we can talk through it then.

**Comment by Brennan1994** [comment]
My Hero!

File: HEC.FDA.Model/paireddata/CurveMetaData.cs [hunk 1]
```diff
@@ -117,9 +117,11 @@ public static CurveMetaData ReadFromXML(XElement xElement)
             string isNullTag = Serialization.GetXMLTagFromProperty(metaDataType, nameof(IsNull));
             if (!bool.TryParse(xElement.Attribute(isNullTag)?.Value, out bool isNull))
                 return new CurveMetaData();
-            
-            CurveMetaData curveMetaData = new CurveMetaData(xLabel, yLabel, name, damageCategory, assetCategory);
-            curveMetaData.IsNull = isNull;
+
```
Comment (Brennan1994) [inline HEC.FDA.Model/paireddata/CurveMetaData.cs:120]: I appreciate you doing these minor cleanups.

File: HEC.FDA.Model/structures/Inventory.cs [hunk 2]
```diff
@@ -335,20 +339,43 @@ public List<DeterministicOccupancyType> SampleOccupancyTypes(IProvideRandomNumbe
...
+                    if (wse[j] != -9999)
+                    {
+                        ConsequenceResult consequenceResult = Structures[i].ComputeDamage(wse[j], deterministicOccupancyType, PriceIndex, analysisYear);
+                        structureParallelCollection[j,i] = (consequenceResult.StructureDamage);
+                        contentParallelCollection[j, i] = (consequenceResult.ContentDamage);
+                        otherParallelCollection[j, i] = (consequenceResult.OtherDamage);
+                        vehicleParallelCollection[j, i] = (consequenceResult.VehicleDamage);
+                    }
+                }
+                Thread.Sleep(0);
```
Comment (Brennan1994) [inline HEC.FDA.Model/structures/Inventory.cs:367]: Is this here for any reason? Sleep for 0 miliseconds?

File: HEC.FDA.Statistics/Distributions/Empirical.cs [hunk 3]
```diff
@@ -402,21 +404,21 @@ public static Empirical StackEmpiricalDistributions(List<Empirical> empiricalDis
...
-
                 for (int j = 1; j < empiricalDistributionsForStacking.Count; j++)
                 {
                     stackedValue = addOrSubtract(stackedValue, empiricalDistributionsForStacking[j].InverseCDF(probabilityStep));
                 }
-
                 cumulativeProbabilities[i] = probabilityStep;
                 stackedInvCDFs[i] = stackedValue;
-            }
+                Thread.Sleep(0);
```
Comment (Brennan1994) [inline HEC.FDA.Statistics/Distributions/Empirical.cs:418]: I'm assuming this is intentional, but it doesn't make sense to me. Maybe comment it to explain why you do it?

**Review by Brennan1994 — APPROVED** [review]

**Comment by Brennan1994** [comment]
@rnugent3, feel free to review my changes. I made a few commits at the end here. 

Generally, I'm on board with everything. Looks great!

File: HEC.FDA.Model/structures/Inventory.cs (hunk 2 above)
Comment (rnugent3) [reply HEC.FDA.Model/structures/Inventory.cs:367]: Here's the description of the method parameter in the .net docs "The number of milliseconds for which the thread is suspended. If the value of the millisecondsTimeout argument is zero, the thread relinquishes the remainder of its time slice to any thread of equal priority that is ready to run. If there are no other threads of equal priority that are ready to run, execution of the current thread is not suspended."

**Comment by rnugent3** [comment]
thanks for making the improvement, @Brennan1994 !!

**Pull request merged**